modules/mono/editor/bindings_generator.cpp

```diff
@@ -1892,6 +1892,24 @@ Error BindingsGenerator::_generate_cs_property(const BindingsGenerator::TypeInte
 		p_output.append("static ");
 	}
 
+	if (prop_itype->proxy_name.contains(p_iprop.proxy_name)) {
```

Comment:
I'd prefer to have an improved version incorporated, though I'm not sure of the best way to calculate performance. Are there any built-in timing metrics that can be activated in the binding generator?

Reply:
Oh sorry, I wasn't looking where this code was. I thought this was about existing `new`, I didn't see you were adding it.

So if I understand, you are adding `new` to avoid the warning that happens when properties shadow other properties from the base class, right? If so, I would prefer not to suppress this warning because it could be happening due to bugs in the bindings generation and we'll want to make sure it is noticed and not hidden if it happens.

There's a bit more discussion about this in: https://github.com/godotengine/godot/issues/15763.

AFAIK, the conclusion is that we wanted to rename the properties in Core to avoid the shadowing, it's a bit late now since we already released 4.0 and can't break compat anymore. The best we could do now is suppress the warnings, but I would not want to blanket suppress all warnings since, as explained earlier, some could be a bug.

Reply:
This isn't a blanket suppress though, it adds "new" specifically to the few instances where an inherited class is hiding something from one of their base classes. That wouldn't be a bug in the binding generator, that's just the way it's laid out in the core

Reply:
It essentially acts like one though, because it's doing it automatically. You are assuming that the `_populate_object_type_interfaces` method is implemented correctly.

For example, what if `ClassDB::get_property_list` was returning not just the properties declared in the given class, but also the inherited ones? In this case the generator would be incorrectly generating shadow properties instead of inheriting them from the base class.

And in this case, your check would just see that the property also exists in the base and assume that it's correct to add `new` here, thus hiding a valid warning.

That's why I'm concerned with automatically hiding warnings in a "smart" way like this. The bindings generator is a huge class and I don't fully trust it. You yourself even found a bug with the enum doc generation.

Reply:
My counterpoint would be that a bug in the system would be caught right away, because there's a warning for applying "new" *incorrectly*. You couldn't just apply "new" to everything because that would put the warnings in the quadruple digits

The alternative to warnings being potentially buried from a bug elsewhere in the binding generator is warnings that currently *do* bury other warnings. Suppressing them uniformly would only realistically serve the same result as this specialized "new" implementation, so I wouldn't consider that a reason to exclude it. Data would be obfuscated in either case, so I prefer having a "correct" implementation over another bandage out of fear of other bandages

Reply:
> My counterpoint would be that a bug in the system would be caught right away, because there's a warning for applying "new" _incorrectly_.

If you are referring to [CS0109](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0109). This would only happen if the bug is in the code added by this PR, not if the bug is in the bindings generation. As I explained earlier, this PR only applies the `new` keyword when the property is shadowing another from the base class, essentially hiding the [CS0108](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0108) warnings.

> You couldn't just apply "new" to everything because that would put the warnings in the quadruple digits

I'm not saying to apply `new` to everything. I'm saying we should only use it when we know for sure the shadowing is intended. But this can't be automated.

> Data would be obfuscated in either case, so I prefer having a "correct" implementation over another bandage out of fear of other bandages

I agree that currently the .NET module generates a lot of warnings that should be fixed. Applying the `new` keyword blindly, in my opinion, is not fixing the warnings, it's just hiding them.

Reply:
Hmmm, now I see your point; the fact that they need a "new" prefix at all is the issue. I'm not entirely sure what the alternative would be if it's really too late to make changes to the names, and leaving them as warnings isn't exactly ideal either...

Reply:
After giving it more thought, what if this system was still used but the warnings were instead migrated to `build_assemblies.py` instead? That is, similar to how there's still warnings for using "reserved names" (despite them already being parsed by that point), there could be warnings for not hiding the methods (despite them already being hidden by that point)